algorithms/linfa-trees/src/decision_trees/algorithm.rs

```diff
@@ -196,13 +197,135 @@ impl<F: Float, L: Label + std::fmt::Debug> TreeNode<F, L> {
+            let valid_split_points: Vec<usize> = (0..mask.mask.len())
+                .filter(|&i| mask.mask[sorted_index.sorted_values[i].0])
+                .collect();
+            random_split_idx = rand::thread_rng().gen_range(0..valid_split_points.len());
+        }
+
+        // We start by putting all available observations in the right subtree
+        // and then move the (sorted by `feature_idx`) observations one by one to
+        // the left subtree and evaluate the quality of the resulting split. At each
+        // iteration, the obtained split is compared with `best`, in order
+        // to find the best possible split.
+        // The resulting split will then have the observations with a value of their `feature_idx`
+        // feature smaller than the split value in the left subtree and the others still in the right
+        // subtree
+        for i in 0..mask.mask.len() - 1 {
+            // (index of the observation, value of its `feature_idx` feature)
+            let (presorted_index, mut split_value) = sorted_index.sorted_values[i];
+
+            // Skip if the observation is unavailable in this subtree
+            if !mask.mask[presorted_index] {
+                continue;
+            }
+
+            // Target and weight of the current observation
+            let sample_class = &target[presorted_index];
+            let sample_weight = data.weight_for(presorted_index);
+
+            // Move the observation from the right subtree to the left subtree
+
+            // Decrement the weight on the class for this sample on the right
+            // side by the weight of this sample
+            *right_class_freq.get_mut(sample_class).unwrap() -= sample_weight;
+            weight_on_right_side -= sample_weight;
+
+            // Increment the weight on the class for this sample on the
+            // right side by the weight of this sample
+            *left_class_freq.entry(sample_class.clone()).or_insert(0.0) += sample_weight;
+            weight_on_left_side += sample_weight;
+
+            // Continue if the next value is equal, so that equal values end up in the same subtree
+            if (i < sorted_index.sorted_values.len() - 1)
+                && (sorted_index.sorted_values[i].1 - sorted_index.sorted_values[i + 1].1).abs()
+                    < F::cast(1e-5)
+            {
+                continue;
+            }
+
+            // If the split would result in too few samples in a leaf
+            // then skip computing the quality
+            if weight_on_right_side < hyperparameters.min_weight_leaf
+                || weight_on_left_side < hyperparameters.min_weight_leaf
+            {
+                continue;
+            }
+
+            // Don't need to compute the quality of other arbitrary splits,
+            // will only consider a split past the random split
+            if hyperparameters.random_split && i < random_split_idx {
+                continue;
+            }
```

Comment:
I don't understand this comment/the logic here

Reply:
Ok I now understand it, making it explicit that  this is to skip all indices until we hit the random split idx would be helpful. Something like
```rust
// Skip indices until we hit `random_split_idx`, since for random splits we only need to consider the randomly selected index
```